spec/services/external_upload_manager_spec.rb

```diff
@@ -0,0 +1,220 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ExternalUploadManager do
+  fab!(:user) { Fabricate(:user) }
+  let(:type) { "card_background" }
+  let!(:logo_file) { file_from_fixtures("logo.png") }
+  let!(:pdf_file) { file_from_fixtures("large.pdf", "pdf") }
+  let(:object_size) { 1.megabyte }
+  let(:etag) { "e696d20564859cbdf77b0f51cbae999a" }
+  let(:client_sha1) { Upload.generate_digest(object_file) }
+  let(:sha1) { Upload.generate_digest(object_file) }
+  let(:object_file) { logo_file }
+  let(:metadata_headers) { {} }
+  let!(:external_upload_stub) { Fabricate(:image_external_upload_stub, created_by: user) }
+  let(:upload_base_url) { "https://#{SiteSetting.s3_upload_bucket}.s3.#{SiteSetting.s3_region}.amazonaws.com" }
+
+  subject do
+    ExternalUploadManager.new(external_upload_stub)
+  end
+
+  before do
+    SiteSetting.authorized_extensions += "|pdf"
+    SiteSetting.max_attachment_size_kb = 210.megabytes / 1000
+
+    setup_s3
+    stub_head_object
+    stub_download_object_filehelper
+    stub_copy_object
+    stub_delete_object
+  end
+
+  describe "#can_promote?" do
+    it "returns false if the external stub status is not created" do
+      external_upload_stub.update!(status: ExternalUploadStub.statuses[:uploaded])
+      expect(subject.can_promote?).to eq(false)
+    end
+  end
+
+  context "when stubbed upload is < DOWNLOAD_LIMIT (small enough to download + generate sha)" do
```

Comment:
Let's group this under `describe "#promote_to_upload!"` so it is clearer which method we are  testing here?